android/sdk/src/main/java/com/taobao/weex/ui/view/gesture/WXGesture.java

```diff
@@ -309,12 +320,14 @@ public boolean onTouch(View v, MotionEvent event) {
           break;
         case MotionEvent.ACTION_MOVE:
           result |= handleMotionEvent(LowLevelGesture.ACTION_MOVE, event);
+          mIsTouchEventConsumed = mIsTouchEventConsumed || result;
           break;
         case MotionEvent.ACTION_UP:
         case MotionEvent.ACTION_POINTER_UP:
           finishDisallowInterceptTouchEvent(v);
           result |= handleMotionEvent(LowLevelGesture.ACTION_UP, event);
           result |= handlePanMotionEvent(event);
+          mIsTouchEventConsumed = mIsTouchEventConsumed || result;
           break;
         case MotionEvent.ACTION_CANCEL:
           finishDisallowInterceptTouchEvent(v);
```

Comment:
click event should not trigger when any of gesture events (start/move)  fired. So `ACTION_CANCEL` will not reset flag.